src/snowflake/snowpark/modin/plugin/extensions/index.py

```diff
@@ -440,7 +476,7 @@ def unique(self, level: Hashable | None = None) -> Index:
             raise IndexError(
                 f"Too many levels: Index has only 1 level, {level} is not a valid level number."
             )
-        return Index(
+        return self.__constructor__(
```

Comment:
CMIIW, since init or new will return DatetimeIndex if the type is datetime64 like, then we could always use Index(..) here right?

Reply:
Yes, that's right.